src/Mono.Android/Android.Telephony/CellInfo.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using Android.Runtime;
+using Java.Interop;
+
+namespace Android.Telephony {
+
+	public partial class CellInfo {
+
+		static Delegate cb_getCellIdentity;
+#pragma warning disable 0169
+		static Delegate GetGetCellIdentityHandler ()
+		{
+			if (cb_getCellIdentity == null)
+				cb_getCellIdentity = JNINativeWrapper.CreateDelegate ((Func<IntPtr, IntPtr, IntPtr>) n_GetCellIdentity);
+			return cb_getCellIdentity;
+		}
+
+		static IntPtr n_GetCellIdentity (IntPtr jnienv, IntPtr native__this)
+		{
+			Android.Telephony.CellInfo __this = global::Java.Lang.Object.GetObject<Android.Telephony.CellInfo> (jnienv, native__this, JniHandleOwnership.DoNotTransfer);
+			return JNIEnv.ToLocalJniHandle (__this.CellIdentity);
+		}
+#pragma warning restore 0169
+
+		public unsafe virtual Android.Telephony.CellIdentity CellIdentity {
+			// Metadata.xml XPath method reference: path="/api/package[@name='android.telephony']/class[@name='CellInfo']/method[@name='getCellIdentity' and count(parameter)=0]"
+			[Register ("getCellIdentity", "()Landroid/telephony/CellIdentity;", "GetGetCellIdentityHandler", ApiSince = 30)]
+			get {
+				const string __id = "getCellIdentity.()Landroid/telephony/CellIdentity;";
```

Comment:
Yes, it can be executed.  Consider:

```csharp
CellInfo     cell     = new CellInfoCdma ();
CellIdentity identity = cell.CellIdentity;
```

`CellInfoCdma` *does not `override`* the `CellInfo.CellIdentity` property.  It *can't*, because in order to do so it would need to change the property type, and changing the property type would be a compatibility break.

Now, a downside to this approach is that if you build your app against API-R and use the `CellInfo.CellIdentity` property, then run the app on e.g. an API-29 device, `cell.CellIdentity` will *throw*, because API-29 doesn't have a `CellInfo.getCellIdentity()` method.

What I don't know is if this behavior is consistent with Java.  I assume it would be, but haven't tested.  (Could someone please test this? :-)

Even if this behavior is consistent with Java, it's kinda "less than ideal", because there *is* a `CellInfoCdma.CellIdentity` property which *does* work on API-29, so why should this throw?

If we think that `cell.CellIdentity` *should* work on API-29, then we need an indirection:

```csharp
partial class CellIdentity {
    public virtual Android.Telephony.CellIdentity CellIdentity {
        [Register (...)]
        get => _GetCellIdentityCore ();

    internal virtual Android.Telephony.CellIdentity _GetCellIdentityCore ()
    {
        const string __id = "getCellIdentity.()Landroid/telephony/CellIdentity;";
        try {
            var __rm = _members.InstanceMethods.InvokeVirtualObjectMethod (__id, this, null);
            return global::Java.Lang.Object.GetObject<Android.Telephony.CellIdentity> (__rm.Handle, JniHandleOwnership.TransferLocalRef);
        }
        catch (Java.Lang.NoSuchMethodError) {
            throw new Java.Lang.AbstractMethodError (__id);
        }
    }
}

sealed partial class CellIdentityCdma {
    public new Android.Telephony.CellIdentityCdma CellIdentity { get; }

    internal override Android.Telephony.CellIdentity _GetCellIdentityCore () => CellIdentity;
}
```